fetch.bs

```diff
@@ -6710,6 +6748,364 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+request exceeds the reporting-origin quota for <code>https://a.example.com</code>, and would throw.
+<pre><code class=lang-javascript>
+  fetchLater("https://a.example.com", {method: "POST", body: a_64kb_body});
+  fetchLater("https://b.example.com", {method: "POST", body: a_64kb_body});
+  fetchLater("https://a.example.com");
+</code></pre>
+
+<p>Same-origin same-agent subframes share the quota of their parent. However, cross-origin or
+cross-agent iframes only receive 8kb of quota by default. So in the following example, the first 3
+calls would succeed and the last one would throw.
+<pre><code class=lang-javascript>
+  // In main page
+  fetchLater("https://a.example.com", {method: "POST", body: a_64kb_body});
+
+  // In same-origin iframe
+  fetchLater("https://b.example.com", {method: "POST", body: a_64kb_body});
+
+  // In cross-origin iframe at https://frame.example.com
+  fetchLater("https://a.example.com", {body: a_5kb_body});
+  fetchLater("https://a.example.com", {body: a_12kb_body});
+</code></pre>
+
+<p>To make the previous example not throw, the top-level {{Document}} needs to delegate some of its
+quota to <code>https://frame.example.com</code>, for example by serving the following header:
+<pre><code>Permissions-Policy: deferred-fetch=(self "https://frame.example.com")</code></pre>
+
+</div>
+
+
+<p>This specification defined a <a>policy-controlled feature</a> identified by the string
+<dfn for=PermissionPolicy enum-value>"deferred-fetch"</dfn>. Its
+<a for="policy-controlled feature">default allowlist</a> is "<code>self</code>".
+
+<p>This specification defined a <a>policy-controlled feature</a> identified by the string
+<dfn for=PermissionPolicy enum-value>"deferred-fetch-minimal"</dfn>. Its
+<a for="policy-controlled feature">default allowlist</a> is "<code>*</code>".
+
+<p>The <dfn>optional subframe deferred-fetch quota</dfn> is 64 kibibytes.
+<p>The <dfn>minimal subframe deferred-fetch quota</dfn> is 8 kibibytes.
+<p>The <dfn>deferred-fetch delegated quota</dfn> is 128 kibibytes.
+
+<div algorithm>
+<p>Two {{Document}}s <var>a</var> and <var>b</var> are said to
+<dfn>share deferred-fetch quota</dfn> if <var>a</var>'s
+<a>relevant agent</a> is <var>b</var>'s <a>relevant agent</a>, and <var>a</var>'s
+<a for=Document>URL</a>'s <a for=url>origin</a> is <a>same origin</a> with <var>b</var>'s
+<a for=Document>URL</a>'s <a for=url>origin</a>.
+</div>
+
+
+<div algorithm>
+<p>To get the <dfn>available deferred-fetch quota</dfn> given a {{Document}}
+<var>document</var> and an <a for=/>origin</a>-or-null <var>origin</var>:
+
+<ol>
+ <li><p>Let <var>quota</var> be zero.
+
+ <li><p>Let <var>quotaForOrigin</var> be 64 kibibytes.
+
+ <li>
+  <p>For each <var>otherNavigable</var> in <var>document</var>'s <a>node navigable</a>'s
+  <a for=navigable>top-level traversable</a>'s <a>inclusive descendant navigables</a>:
+
+  <p class=note>This algorithm iterates over the entire navigable tree. It accumulates quota from
```

Comment:
```
This algorithm iterates over the entire navigable tree.
```

Given the following example

```
root (a.com) -> frame_1 (a.com)
             -> frame_2 (a.com)
             -> frame_3 (b.com)
             -> frame_4 (b.com)
```

Let document be `frame_3` on its navigation, running the algorithm gives the following

* `quotaForOrigin` = 64 kb
* `quota` = 8*2 kb, as the following step is run twice, on `frame_3` and `frame_4`:
  
   ```
   Otherwise, if otherContainer’s subframe reserved deferred-fetch quota is minimal 
   subframe deferred-fetch quota and otherDocument is allowed to use the policy- 
    controlled feature "deferred-fetch-minimal", then increment quota by minimal 
   subframe deferred-fetch quota.
   ```
* return 64 kb (quotaForOrigin)

But the quota for `frame_3` (or `frame_4`) should just be `8kb` by default?

Reply:
It's a good example! Yea, they share quota and each receives 8kb, so they can share 16kb of total delegated quota.

Reply:
How can b.com reserve itself quota? It would only receive quota if a.com created 16 iframes with b.com

Reply:
1 and 2 receive 64kb upon navigating, which is reset to 0 when the document is created.

Reply:
If frames are created sequentialy, does that mean frame_3 receives 8kb and frame_4 receives 16kb? Or will frame_3 get an updated 16kb quota?